processing/src/main/java/io/druid/segment/filter/SelectorFilter.java

```diff
@@ -31,25 +40,68 @@
 {
   private final String dimension;
   private final String value;
+  private final ExtractionFn extractionFn;
 
   public SelectorFilter(
       String dimension,
-      String value
+      String value,
+      ExtractionFn extractionFn
   )
   {
     this.dimension = dimension;
     this.value = value;
+    this.extractionFn = extractionFn;
   }
 
   @Override
   public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   {
-    return selector.getBitmapIndex(dimension, value);
+    if (extractionFn == null) {
+      return selector.getBitmapIndex(dimension, value);
+    } else {
+      final List<Filter> filters = makeFilters(selector);
+      if (filters.isEmpty()) {
+        return selector.getBitmapFactory().makeEmptyImmutableBitmap();
+      }
+      return new OrFilter(makeFilters(selector)).getBitmapIndex(selector);
+    }
   }
 
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    return factory.makeValueMatcher(dimension, value);
+    if (extractionFn == null) {
+      return factory.makeValueMatcher(dimension, value);
+    } else {
+      return factory.makeValueMatcher(
+          dimension, new Predicate<String>()
+          {
+            @Override
+            public boolean apply(String input)
+            {
+              // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+              return value.equals(Strings.nullToEmpty(extractionFn.apply(Strings.emptyToNull(input))));
```

Comment:
This feels like a lot of hoops to jump through.

The ExtractionFn contract probably needs clarified on what goes in. Just looking at a few spot cases it is not obvious things like the JavaScript extraction fn handles null inputs in a proper way.

Given that the output is always null and never empty (by method contract) it seems like we should be able to eliminate the nullToEmpty... and if the ability for ExtractionFn.apply to accept either null xor empty in the method contract, that should help clarify the need for the emptyToNull.

Some ExtractionFuncions seem to handle null input explicitly, and others do not seem to do so, which may be either a waiting bug or an unneeded check.

Anyways, such a change would be outside the scope of this PR and the approach here is fine for now, but please make sure `value` has a null check somewhere before it reaches here.